### PR TITLE
[#162415][Daily Rate] Charge for daily rated instruments

### PR DESCRIPTION
**Pull request by joaquinco**
# Notes

- Refactor `PricePolicies::TimeBasedPriceCalculator` by splitting the logic based on the pricing strategy
- Fix formatting on spec file
- Add tests for daily rate pricing calculation

### CODE REVIEW

File: spec/services/price_policies/time_based_price_calculator_spec.rb [hunk 1]
```diff
@@ -225,6 +225,14 @@
...
             is_expected.to eq(cost: 120, subsidy: 0)
           end
+
+          it "calls the correct strategy" do
+            expect_any_instance_of(
+              PricePolicies::Strategy::PerMinute
+            ).to receive(:calculate)
+
+            subject
+          end
```
Comment (LeticiaErrandonea) [inline spec/services/price_policies/time_based_price_calculator_spec.rb:235]: Given in line 169 we're defining `subject` using `PricePolicies::Strategy::PerMinute`, is this spec useful?

File: spec/services/price_policies/time_based_price_calculator_spec.rb [hunk 2]
```diff
@@ -328,4 +354,51 @@
       end
     end
   end
+
+  context "when instrument has daily rate pricing" do
```
Comment (LeticiaErrandonea) [inline spec/services/price_policies/time_based_price_calculator_spec.rb:358]: Can we add a spec for edge cases? Even when it should be the same, I think it's worth it to test a reservation that starts at 00:00

**Review by LeticiaErrandonea — COMMENTED** [review]
Looking good!

Left some comments/questions for specs

File: spec/services/price_policies/time_based_price_calculator_spec.rb (hunk 1 above)
Comment (joaquinco) [reply spec/services/price_policies/time_based_price_calculator_spec.rb:235]: Line 169 is just defining a let like declaration named `discount`, I had to change that due to the refactor

File: spec/services/price_policies/time_based_price_calculator_spec.rb (hunk 2 above)
Comment (joaquinco) [reply spec/services/price_policies/time_based_price_calculator_spec.rb:358]: Sure

**Review by LeticiaErrandonea — APPROVED** [review]
LGTM!

Thanks for the changes!
